aspnetcore/data/ef-rp/concurrency.md

```diff
@@ -0,0 +1,333 @@
+---
+title: Razor Pages with EF Core - Concurrency - 8 of 8
+author: rick-anderson
+description: This tutorial shows how to handle conflicts when multiple users update the same entity at the same time.
+keywords: ASP.NET Core,Entity Framework Core,concurrency
+ms.author: riande
+manager: wpickett
+ms.date: 11/15/2017
+ms.topic: get-started-article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: data/ef-rp/concurrency
+---
+
+en-us/
+
+# Handling concurrency conflicts - EF Core with Razor Pages (8 of 8)
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT), [Tom Dykstra](https://github.com/tdykstra), and  [Jon P Smith](https://twitter.com/thereformedprog)
+
+[!INCLUDE[about the series](../../includes/RP-EF/intro.md)]
+
+This tutorial shows how to handle conflicts when multiple users update an entity concurrently (at the same time). If you run into problems you can't solve, download the [completed app for this stage](https://github.com/aspnet/Docs/tree/master/aspnetcore/data/ef-rp/intro/samples/StageSnapShots/cu-part8).
+
+## Concurrency conflicts
+
+A concurrency conflict occurs when:
+
+* A user navigates to the edit page for an entity.
+* Another user updates the same entity before the first user's change is written to the DB.
+
+If concurrency detection is not enabled, when concurrent updates occur:
+
+* The last update wins. That is, the last update values are saved to the DB.
+* The first of the current updates are lost. That's generally not what you want.
+
+### Pessimistic concurrency (locking)
+
+DB locking, also called pessimistic concurrency, is a concurrency approach. For example, before a row is read from the db, the app can request a lock for read-only or for update access. If a row is locked for update access:
+
+* Other users can't lock the row for read-only access. That is, other users can't read the data.
+* Other users can't lock the row for update access.
```

Comment:
This seems a bit confusing to me :smile: Is this definition of what "locked for update access" means coming from somewhere else? E.g. update locks in SQL Server are compatible with shared locks, which is what I believe you are referring to with "locked for read-only acesss".